crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -1027,15 +1162,48 @@ pub trait EntityCommand<Marker = ()>: Send + 'static {
     /// footprint than `(Entity, Self)`.
     /// In most cases the provided implementation is sufficient.
     #[must_use = "commands do nothing unless applied to a `World`"]
-    fn with_entity(self, entity: Entity) -> impl Command
+    fn with_entity(self, entity: Entity, failure_mode: FailureMode) -> impl Command
     where
         Self: Sized,
     {
-        move |world: &mut World| self.apply(entity, world)
+        move |world: &mut World| {
+            if world.entities.contains(entity) {
+                self.apply(entity, world);
+            } else {
+                match failure_mode {
+                    FailureMode::Ignore => (),
+                    FailureMode::Log => {
+                        info!("Could not execute EntityCommand because its Entity {entity:?} was missing");
```

Comment:
You may want to use Display formatting. There is an open issue about consistently preferring Display for Entity.